tools/compile_tf_graph.py

```diff
@@ -368,11 +466,12 @@ def set_state_var_final_value(self, name, final_value):
     """
     self.state_vars[name].set_final_value(final_value)
 
-  def create_state_vars_recursive(self, name_prefix, initial_values, data_shape=None):
+  def create_state_vars_recursive(self, name_prefix, initial_values, data_shape=None, choice_dependent=None):
```

Comment:
Why is this `choice_dependent` needed?

Reply:
these state vars which can be infered from choice does not need to be saved into Sprint CPU

Reply:
I don't understand. Everything can be inferred from choice and current state?